tests/test_gerrit.py

```diff
@@ -0,0 +1,407 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2015-2017 Bitergia
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, 51 Franklin Street, Fifth Floor, Boston, MA 02110-1335, USA.
+#
+# Authors:
+#     Valerio Cosentino <valcos@bitergia.com>
+#
+
+import collections
+import datetime
+import os
+import subprocess
+import sys
+import unittest
+
+import pkg_resources
+
+# Hack to make sure that tests import the right packages
+# due to setuptools behaviour
+sys.path.insert(0, os.path.join(os.path.dirname(os.path.abspath(__file__)), ".."))
+pkg_resources.declare_namespace('perceval.backends')
+
+from perceval.backend import BackendCommandArgumentParser
+from perceval.errors import BackendError
+from perceval.utils import DEFAULT_DATETIME
+
+from perceval.backends.core.gerrit import (CATEGORY_REVIEW, MAX_REVIEWS, PORT,
+                                           Gerrit,
+                                           GerritCommand,
+                                           GerritClient)
+
+from base import TestCaseBackendArchive
+
+
+GERRIT_REPO = "gerrit.acme.org"
+GERRIT_USER = "user"
+
+VERSION_UNKNOWN = 'data/gerrit/gerrit_version_unknown'
+VERSION_214 = 'data/gerrit/gerrit_version_214'
+REVIEWS_PAGE_1 = 'data/gerrit/gerrit_reviews_page_1'
+REVIEWS_PAGE_2 = 'data/gerrit/gerrit_reviews_page_2'
+REVIEWS_PAGE_3 = 'data/gerrit/gerrit_reviews_page_3'
+
+CMD_VERSION = "ssh  -p 29418 user@gerrit.acme.org gerrit  version "
+CMD_REVIEWS_1 = "ssh  -p 29418 user@gerrit.acme.org gerrit  query limit:2 " \
+                "'(status:open OR status:closed)' --all-approvals --comments --format=JSON --start=0"
+CMD_REVIEWS_2 = "ssh  -p 29418 user@gerrit.acme.org gerrit  query limit:2 " \
+                "'(status:open OR status:closed)' --all-approvals --comments --format=JSON --start=2"
+CMD_REVIEWS_3 = "ssh  -p 29418 user@gerrit.acme.org gerrit  query limit:2 " \
+                "'(status:open OR status:closed)' --all-approvals --comments --format=JSON --start=4"
+
+
+def read_file(filename, mode='r'):
+    with open(os.path.join(os.path.dirname(os.path.abspath(__file__)), filename), mode) as f:
+        content = f.read()
+    return content
+
+
+class MockedCommand():
```

Comment:
I see the point of this object but I don't think is the best way to mock a function. I suggest to use `unittest.mock` module instead with the help of a class similar to this one.
This call `setattr(obj, member, mock)` can propagate side-effects when something fails in a test.